### PR TITLE
hardware/cpu: add `virtualized?` check

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
This PR adds a check for virtualization, so we no longer have to use the `ENV["HOMEBREW_GITHUB_ACTIONS"]` hack to work around CI failures.

### CODE REVIEW

**Comment by p-linnane** [comment]
Thank you to @Bo98 for the idea & guidance here 🙇‍♂️

File: Library/Homebrew/hardware.rb [hunk 1]
```diff
@@ -156,6 +156,11 @@ def big_endian?
         [1].pack("I") == [1].pack("N")
       end
 
+      def virtualized?
+        puts "Detecting virtualization is not supported on Linux."
```
Comment (Bo98) [inline Library/Homebrew/hardware.rb:160]: We probably don't want this outputting things to stdout

**Review by Bo98 — APPROVED** [review]
Thanks!

**Comment by MikeMcQuaid** [comment]
Good idea @p-linnane!

**Comment by MikeMcQuaid** [comment]
> so we no longer have to use the `ENV["HOMEBREW_GITHUB_ACTIONS"]` hack to work around CI failures.

@p-linnane although: please note on this that this should never be used in `test do` blocks as those should be consistent between CI and run on contributors' machines. No sense in having `brew test`s that aren't run in CI. For build differences: fair enough.

**Comment by p-linnane** [comment]
This is primarily to deal with things like OpenCL, which doesn't work on ephemeral ARM, or other GPU related things. Here's an example of a test update I made yesterday, and @Bo98 weighed in on adding this check: https://github.com/Homebrew/homebrew-core/pull/160229

**Comment by Bo98** [comment]
Yeah OpenCL not working is not a CI limitation but rather a virtualisation limitation. So this is a more correct check than HOMEBREW_GITHUB_ACTIONS in those situations as it will be correct should anyone run the test in a local VM.

**Comment by MikeMcQuaid** [comment]
@p-linnane @Bo98 Gotcha, thanks, seems reasonable. I guess what I'm suggesting is that we avoid `brew test` lines that are never exercised in CI but that's not the case here 👍🏻

**Pull request merged**